cranelift/fuzzgen/src/lib.rs

```diff
@@ -56,14 +56,14 @@ where
                 .params
                 .iter()
                 .map(|p| {
-                    let imm64 = match p.value_type {
-                        I8 => self.u.arbitrary::<i8>()? as i64,
-                        I16 => self.u.arbitrary::<i16>()? as i64,
-                        I32 => self.u.arbitrary::<i32>()? as i64,
-                        I64 => self.u.arbitrary::<i64>()?,
+                    let imm = match p.value_type {
+                        I8 => self.u.arbitrary::<i8>()? as i128,
+                        I16 => self.u.arbitrary::<i16>()? as i128,
+                        I32 => self.u.arbitrary::<i32>()? as i128,
+                        I64 => self.u.arbitrary::<i64>()? as i128,
```

Comment:
I guess we could start generating signatures with `i128`s here? Or do you want to add that later?

Reply:
I think I'd need to add support elsewhere as well, I think its better to do it all together later.